### PR TITLE
:sparkles: Custom filter placeholder text for data table

### PR DESCRIPTION
**Pull request by fraghag**
This has been asked for by multiple people in the community:
https://community.plotly.com/t/custom-text-in-datatable-filter-cell/34950/1

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] Add setting for custom data table column filter placeholder text
   -  [x] Update the docs
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### optionals

- [ ] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follows
    -  [ ] this GitHub [#PR number]() updates the dash docs
    -  [ ] here is the show and tell thread in Plotly Dash community

### CODE REVIEW

File: components/dash-table/tests/selenium/test_filter.py [hunk 1]
```diff
@@ -301,3 +301,40 @@ def test_filt003_sensitivity(test, filter_options, column_filter_options):
...
+
+
+@pytest.mark.parametrize(
+    "column_placeholder_setting,table_placeholder_setting,expected_placeholder",
+    [
+        ("abc", None, "abc"),
+        (None, "def", "def"),
+        ("gah", "ijk", "gah"),
+        ("", None, ""),
+        (None, None, "filter data..."),
```
Comment (alexcjohnson) [inline components/dash-table/tests/selenium/test_filter.py:313]: Beautiful set of test cases! One additional case I'm curious about though: what if only `filter_options.case` is set at the column level, does the table-level `filter_options.placeholder` still make it through? ie does the column-level setting merge with or replace the table-level setting? Merging would be preferred.

Vice versa too, I suppose: does table-level `case` survive if only `placeholder` is supplied by the column?

File: components/dash-table/src/dash-table/dash/DataTable.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import * as R from 'ramda';
-import React, {Component, Suspense} from 'react';
+import React, { Component, Suspense } from 'react';
 import PropTypes from 'prop-types';
-import {asyncDecorator} from '@plotly/dash-component-plugins';
+import { asyncDecorator } from '@plotly/dash-component-plugins';
```
Comment (alexcjohnson) [inline components/dash-table/src/dash-table/dash/DataTable.js:4]: The lint job failed, I think our settings actually prohibit these spaces. `npm run format` from `components/dash-table` should sort this out.

**Comment by alexcjohnson** [comment]
Thanks @fraghag - very nicely done. Aside from my two minor comments above, just a changelog entry and I think we'll be good to go!

File: components/dash-table/tests/selenium/test_filter.py (hunk 1 above)
Comment (fraghag) [reply components/dash-table/tests/selenium/test_filter.py:323]: This is handled [here](https://github.com/plotly/dash/blob/2bd83a98322f95243359d846ec9e97a5b542a36e/components/dash-table/src/dash-table/dash/Sanitizer.ts#L85). Added the usecase to the tests for both placeholder_text and case.

File: components/dash-table/src/dash-table/dash/DataTable.js (hunk 2 above)
Comment (fraghag) [reply components/dash-table/src/dash-table/dash/DataTable.js]: It was my VSCode living it's own life. Fixed it now with npm, thanks!

**Comment by fraghag** [comment]
> Thanks @fraghag - very nicely done. Aside from my two minor comments above, just a changelog entry and I think we'll be good to go!

Fixed the comments now. But regarding the changelog, I don't understand where in the Changelog file I'm supposed to add this, I only see entries from already released versions. I was looking form info about this in the CONTRIBUTING file. Could you give me some pointers?

File: components/dash-table/tests/selenium/test_filter.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import dash
 
 from dash.dash_table import DataTable
+from pyparsing import col
```
Comment (alexcjohnson) [inline components/dash-table/tests/selenium/test_filter.py:4]: Didn't end up being used? [Linter flagged](https://app.circleci.com/pipelines/github/plotly/dash/4049/workflows/ec453535-6f7e-4a75-a052-514add2cafb1/jobs/66999):

```
tests/selenium/test_filter.py:4:1: F401 'pyparsing.col' imported but unused
```

**Comment by alexcjohnson** [comment]
Re: changelog - yeah we should describe that better, good point. You can add a new section at the top, right before the `[2.6.2]` heading:

```
## Unreleased

### Added

- [#2261](https://github.com/plotly/dash/pull/2261) 1-2 sentence description of the feature
```

File: components/dash-table/tests/selenium/test_filter.py (hunk 3 above)
Comment (fraghag) [reply components/dash-table/tests/selenium/test_filter.py]: Fixed!

**Review by alexcjohnson — APPROVED** [review]
💃 Excellent work, thanks again for the PR @fraghag!

**Pull request merged**